run_linux.go

```diff
@@ -340,7 +340,21 @@ func runSetupBuiltinVolumes(mountLabel, mountPoint, containerDir string, builtin
 			}
 			initializeVolume = true
 		}
-		stat, err := os.Stat(srcPath)
+
+		stat, err := os.Lstat(srcPath)
+
+		// From a cursory test, Docker (19.03) follows the symlink and mounts
+		// the volume at the destination. We replicate such behavior here.
+		if err == nil && stat.Mode()&os.ModeSymlink != 0 {
+			targetVolume, linkErr := os.Readlink(srcPath)
+			if linkErr != nil {
+				return nil, errors.Wrapf(linkErr, "error following source symlink %q for volume %q", srcPath, volume)
+			}
+
+			srcPath = filepath.Join(mountPoint, targetVolume)
+			stat, err = os.Lstat(srcPath)
+		}
```

Comment:
@nalind @rhatdan should we use a `securejoin.SecureJoin()` here?  Or do you think  the prior checks are sufficient?  My symlink burn marks are throbbing a bit.....

Reply:
Yes, the prior checks are not sufficient and this PR is flawed as well. I have sent a private email to @rhatdan about this. May I copy the email to you?

Reply:
Yes please do,  I was on PTO and have not cought up.